frontend/src/metabase/lib/dom.js

```diff
@@ -349,13 +356,35 @@ const getOrigin = url => {
 const getLocation = url => {
   try {
     const { pathname, search, hash } = new URL(url, window.location.origin);
+    const sitePath = getSitePath();
+    const pathNameWithoutSubPath =
+      sitePath === "/"
+        ? pathname
+        : pathname.replace(
+            // Only replace the path from the start to avoid matching the string elsewhere
+            new RegExp(`^${ensureNoTrailingSlash(sitePath)}`),
+            "",
+          );
     const query = querystring.parse(search.substring(1));
-    return { pathname, search, query, hash };
+    return {
+      pathname: pathNameWithoutSubPath,
```

Comment:
This change is needed when navigating with client-router and Metabase is hosted on a subpath e.g. http://localhost:3000/metabase instead of just http://localhost:3000

Otherwise, the pathname would contain the Metabase subpath as well. This will cause the link to contain the subpath twice e.g. `http://localhost:3000/metabase/metabase/question/1`

Reply:
The change in unclear. Do native `react-router` location objects also exclude `subpath`?

Reply:
@ranquild Apparently we use `basename` , in this case it's `"/metabase"`


https://github.com/metabase/metabase/blob/e0e0646201eeb5fd1d97a4b3121a2fe77b1f00d3/frontend/src/metabase/app.js#L60C19-L65

That's why the `pathname` returned from this function shouldn't contain the subpath `/metabase` again.

Reply:
Great investigation, it's clear now